### ISSUE TITLE
Bypass Rail Item

### ISSUE THREAD

**Issue by lhwi** [jonafanho/Minecraft-Transit-Railway#650]
Currently, if I want to create a bypass route where trains will actually use it, I have to create the line in either:

1. Create separate line with no mergers with the mainline 

2. Have the platform for the bypass station not be on the same side/direction as the next station*. 

3. Create a “platform node” on the bypass route and have the train come to a stop.

Creating a rail item that denotes that this is a bypass rail would help. Maybe take the signals track for inspiration. Allow the dashboard to designate mainline / branch line / alternates

*If I want to bypass a platform on the left, the next station for my train must be on the right. If the next station for my train is on the left, on the same side as the station I wish to bypass, the train would take the left and possibly get stuck behind the stopped train, defeating the whole purpose of the bypass rail.

**Comment by Ultimate-USA-YT** [comment]
Train Pathfinding Preffers Higher Speeds

**Comment by lhwi** [comment]
> Train Pathfinding Preffers Higher Speeds

No it does not, I’ve tested it. It goes by direction first. If the station is to the left of its current location, it will take the first left, even if there’s a clear, faster bypass to the right.

**Comment by Ultimate-USA-YT** [comment]
Try Making the Branch at a Lower Speed

**Comment by kettle-7** [comment]
It's not about distance, nor the actual speed it takes for the train to go
down the rail, its about the speed limit. If the speed limit on the branch
line is lower than the mainline then the train will only take the branch
line if it has to.

**Comment by Ultimate-USA-YT** [comment]
> > It's not about distance, nor the actual speed it takes for the train to go down the rail, its about the speed limit. If the speed limit on the branch line is lower than the mainline then the train will only take the branch line if it has to.
> 
> If your line is straight, then it is true. However, if the next station is offset to the left/right, it will take the first left/right regardless of speed limit.

Just do what he States

**Comment by EpicPuppy613** [comment]
> > It's not about distance, nor the actual speed it takes for the train to go down the rail, its about the speed limit. If the speed limit on the branch line is lower than the mainline then the train will only take the branch line if it has to.
> 
> If your line is straight, then it is true. However, if the next station is offset to the left/right, it will take the first left/right regardless of speed limit.

this is not true, the pathfinding code prefers a higher speed limit track **over** the direction the rail is going

**Comment by Ultimate-USA-YT** [comment]
> > > It's not about distance, nor the actual speed it takes for the train to go down the rail, its about the speed limit. If the speed limit on the branch line is lower than the mainline then the train will only take the branch line if it has to.
> > 
> > 
> > If your line is straight, then it is true. However, if the next station is offset to the left/right, it will take the first left/right regardless of speed limit.
> 
> this is not true, the pathfinding code prefers a higher speed limit track **over** the direction the rail is going

EpicPuppy613 is Correct here

**Comment by jonafanho** [comment]
Pathfinding has been improved in 4.0.0. Please check if this issue has been resolved, thanks.

**Comment by jonafanho** [comment]
Waypoints are tracked in #665
